draft-ietf-quic-qpack.md

```diff
@@ -164,6 +150,69 @@ unmodified.
 A string literal without a prefix length noted is an 8-bit prefix string literal
 and follows the definitions in [RFC7541] without modification.
 
+## Indexing
+
+Entries in the QPACK static and dynamic tables are addressed separately.
+
+Entries in the static table have the same indices at all times.  The static
+table is defined in Appendix A of {{!RFC7541}}. Note that because HPACK did not
+use zero-based references, there is no value at index zero of the static table.
+
+Entries are inserted into the dynamic table over time.  Each entry possesses
+both an absolute index which is fixed for the lifetime of that entry and a
+relative index which changes over time based on the context of the reference.
+The first entry inserted has an absolute index of "1"; indices
+increase sequentially with each insertion.
+
+On the control stream, a relative index of "0" always refers to the most
+recently inserted value in the dynamic table.  Note that this means the
+entry referenced by a given relative index can change while interpreting
+a HEADERS frame as new entries are inserted.
+
+~~~~~ drawing
+    +---+---------------+-------+
+    | n |      ...      | d + 1 |  Absolute Index
+    + - +---------------+   -   +
+    | 0 |      ...      | n-d-1 |  Relative Index
+    +---+---------------+-------+
+      ^                     |
+      |                     V
+Insertion Point         Dropping Point
+
+n = count of entries inserted
+d = count of entries dropped
+~~~~~
+{: title="Dynamic Table Indexing - Control Stream"}
+
+Because frames from request streams can be delivered out of order with
+instructions on the control stream, relative indices are based on the Base Index
+(see {{absolute-index}}) included in the HEADERS or PUSH_PROMISE frame.  The
+Base Index is the absolute index of the entry which has the relative index of
+zero when interpreting the frame.  The relative indices of entries do not change
+while interpreting headers on a request or push stream.
+
+~~~~~ drawing
+    +---+-----+-----+-----+-------+
+    | n | n-1 | n-2 | ... |  d+1  |  Absolute Index
+    +---+-----+  -  +-----+   -   +
+              |  0  | ... | n-d-3 |  Relative Index
+              +-----+-----+-------+
+                 ^
+                 |
+             Base Index
```

Comment:
Since Base Index is an absolute index, would it be more intuitive to have this on the top of the drawing pointing down?